### PR TITLE
Refactor processor

### PR DESCRIPTION
**Pull request by AdoAdoAdo**

### CODE REVIEW

File: testscommon/economicsmocks/economicsHandlerStub.go [hunk 1]
```diff
@@ -36,107 +36,138 @@ type EconomicsHandlerMock struct {
 	ComputeGasUsedAndFeeBasedOnRefundValueCalled func(tx process.TransactionWithFeeHandler, refundValue *big.Int) (uint64, *big.Int)
 	ComputeTxFeeBasedOnGasUsedCalled             func(tx process.TransactionWithFeeHandler, gasUsed uint64) *big.Int
 	ComputeGasLimitBasedOnBalanceCalled          func(tx process.TransactionWithFeeHandler, balance *big.Int) (uint64, error)
+	GenesisTotalSupplyCalled                     func() *big.Int
 }
 
 // LeaderPercentage -
-func (ehm *EconomicsHandlerMock) LeaderPercentage() float64 {
-	return ehm.LeaderPercentageCalled()
+func (ehm *EconomicsHandlerStub) LeaderPercentage() float64 {
```
Comment (SebastianMarian) [inline testscommon/economicsmocks/economicsHandlerStub.go:43]: ehs instead ehm ?

File: testscommon/gasHandlerStub.go [hunk 2]
```diff
@@ -21,74 +21,74 @@ type GasHandlerMock struct {
 }
 
 // Init -
-func (ghm *GasHandlerMock) Init() {
+func (ghm *GasHandlerStub) Init() {
```
Comment (SebastianMarian) [inline testscommon/gasHandlerStub.go:24]: ghs instead ghm ?

File: process/block/preprocess/rewardTxPreProcessor.go [hunk 3]
```diff
@@ -85,10 +85,12 @@ func NewRewardTxPreprocessor(
...
 	bpp := &basePreProcess{
-		hasher:               hasher,
-		marshalizer:          marshalizer,
-		shardCoordinator:     shardCoordinator,
-		gasHandler:           gasHandler,
+		hasher:      hasher,
+		marshalizer: marshalizer,
+		gasTracker: gasTracker{
+			shardCoordinator: shardCoordinator,
+			gasHandler:       gasHandler,
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply process/block/preprocess/rewardTxPreProcessor.go:92]: it is not needed for RewardsTxPreProcessor, but could be set explicitly to nil

File: testscommon/economicsmocks/economicsHandlerStub.go (hunk 1 above)
Comment (AdoAdoAdo) [reply testscommon/economicsmocks/economicsHandlerStub.go:43]: done

File: testscommon/gasHandlerStub.go (hunk 2 above)
Comment (AdoAdoAdo) [reply testscommon/gasHandlerStub.go:24]: done

**Review by SebastianMarian — DISMISSED** [review]

File: process/block/preprocess/transactions.go [hunk 4]
```diff
@@ -920,256 +924,142 @@ func (txs *transactions) createAndProcessMiniBlocksFromMeV1(
...
-			}
-			numNewTxs += common.AdditionalScrForEachScCallOrSpecialTx
-		}
-
-		if isMaxBlockSizeReached(numNewMiniBlocks, numNewTxs) {
-			log.Debug("max txs accepted in one block is reached",
-				"num txs added", numTxsAdded,
-				"total txs", len(sortedTxs))
+	for _, wtx := range sortedTxs {
+		canAddTx, canContinue, tx := mbBuilder.checkAddTransaction(wtx)
```
Comment (raduchis) [inline process/block/preprocess/transactions.go:951]: are both canAdd and canContinue needed? I think that returning just canContinue should suffice

File: process/block/preprocess/transactions.go [hunk 5]
```diff
@@ -920,256 +924,142 @@ func (txs *transactions) createAndProcessMiniBlocksFromMeV1(
...
-		if isAddressSet {
-			ok = txs.balanceComputation.SubBalanceFromAddress(tx.GetSndAddr(), txMaxTotalCost)
-			if !ok {
-				log.Error("createAndProcessMiniBlocksFromMeV1.SubBalanceFromAddress",
-					"sender address", tx.GetSndAddr(),
-					"tx max total cost", txMaxTotalCost,
-					"err", process.ErrInsufficientFunds)
-			}
-		}
+	// TODO: is this really needed? move this somewhere else
```
Comment (raduchis) [inline process/block/preprocess/transactions.go:1028]: is the TODO still relevant?

File: testscommon/hashingMocks/hasherMock.go [hunk 6]
```diff
@@ -1,4 +1,4 @@
-package mock
```
Comment (raduchis) [inline testscommon/hashingMocks/hasherMock.go:1]: All the functions of the HasherMock are on a HasherMock not on a *HasherMock. Can you please change this?

**Review by raduchis — COMMENTED** [review]
Small questions

File: common/converters_test.go [hunk 7]
```diff
@@ -23,7 +24,7 @@ func TestCalculateHash_NilMarshalizer(t *testing.T) {
 	t.Parallel()
 
 	obj := []byte("object")
-	hash, err := core.CalculateHash(nil, &mock.HasherMock{}, obj)
+	hash, err := core.CalculateHash(nil, &hashingMocks.HasherMock{}, obj)
```
[the remark this answers is not in the document]
Comment (AdoAdoAdo) [reply common/converters_test.go:27]: would leave it like this for now and have another PR to refactor the names for all packages inside testscommon.

File: process/block/preprocess/transactions.go (hunk 5 above)
Comment (AdoAdoAdo) [reply process/block/preprocess/transactions.go:1028]: will remove TODO

File: testscommon/hashingMocks/hasherMock.go (hunk 6 above)
Comment (AdoAdoAdo) [reply testscommon/hashingMocks/hasherMock.go]: done

**Review by SebastianMarian — APPROVED** [review]

File: common/converters_test.go (hunk 7 above)
[the remark this answers is not in the document]
Comment (raduchis) [reply common/converters_test.go:27]: ok!

File: process/block/preprocess/transactions.go (hunk 5 above)
Comment (raduchis) [reply process/block/preprocess/transactions.go]: ok!

File: testscommon/hashingMocks/hasherMock.go (hunk 6 above)
Comment (raduchis) [reply testscommon/hashingMocks/hasherMock.go]: ok!

**Review by raduchis — APPROVED** [review]

**Pull request merged**